Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h

```diff
@@ -2546,10 +2550,10 @@ degree(Face_index f) const
     return count;
 }
 
-template <typename P>
+template <typename P> template< typename F>
 void
 Surface_mesh<P>::
-collect_garbage()
+collect_garbage(F& funct)
```

Comment:
@sloriot Could you you review this, please ? I'm not very comfortable with adding code in there.

Reply:
except from the names that I don't like and the typo, it is OK.

Reply:
And now, @sloriot?

Reply:
@maxGimeno Can you please point to the new lines of code?

Reply:
Looks good